witness/tests/Test/Dscp/Witness/Explorer/Explorer.hs

```diff
@@ -92,13 +92,19 @@ data SpoilVariant
     deriving (Show, Eq, Generic, Enum, Bounded)
 
 -- | Spoils a @'MerkleProof'@ by changing some of its internal nodes
--- TODO: spoil better.
 spoilMerkleProof
-    :: MerkleProof a
+    :: (Eq a, Arbitrary a)
+    => MerkleProof a
     -> Gen (MerkleProof a)
-spoilMerkleProof mproof = do
-    newSig <- arbitrary
-    pure $ mproof { pnSig = newSig }
+spoilMerkleProof (ProofBranch l r) = do
+    var <- choose (0 :: Int, 1)
+    if var == 0
```

Comment:
Why not `var :: Bool`? (if you find this more aesthetically correct, I don't object though)